### ISSUE TITLE
Convenience for attributes on Sink

### ISSUE THREAD

**Issue by patriknw** [akka/akka#30199]
When working on https://github.com/akka/akka-persistence-cassandra/pull/888 I realized that it wasn't obvious how to set a dispatcher all the way to the Sink (and not introduce async boundary).

My first thought, when I didn't realize that the Sink is also involved, was:
```
.map(stuff)
.withAttributes(ActorAttributes.dispatcher(theDispatcher))
.runWith(Sink.ignore)
```

but it should be:
```
.map(stuff)
.toMat(Sink.ignore)(Keep.right)
.withAttributes(ActorAttributes.dispatcher(theDispatcher))
.run()
```

**Comment by johanandren** [comment]
I think it would be confusing if it was an extra parameter to `runWith(sink, attributes)`, at least my intuition says that it is for the sink only rather than the whole graph, and after you have `runWith` the stream is already running so there is no way to add the attribute with an additional method after that `runWith` call.

An additional `run` overload that takes attribute, would be obvious, but it wouldn't help for this case as you would anyway have to switch to the `to(sink)` api from `runWith` then.

**Comment by jxnu-liguobin** [comment]
So, Is this a problem worth improving? If so, I'm interested in it.

**Comment by johanandren** [comment]
I don't think so, because with an overload like `.runWith(sink, attributes)` it would be unclear if the attributes for the sink or the rest of the graph or the combined sink. 

Also `source.via(flow).to(sink).run(attributes)` would not improve much compared to `source.via(flow).to(sink).withAttributes(attributes).run()` which already exists.

It could perhaps be clarified in docs, for example on `runWith` scaladoc entries but I am not entirely sure that is worth adding there.

**Comment by jxnu-liguobin** [comment]
> I don't think so, because with an overload like `.runWith(sink, attributes)` it would be unclear if the attributes for the sink or the rest of the graph or the combined sink.

Thank you for your reply.  I think so, too 👍🏻

**Comment by He-Pin** [comment]
I think this is more or less the reactor-core 's subscibeOn/publishing thing.

How about add a overload method runOn(xxxSink, xxxDispatcher) ?
